### PR TITLE
Memfiles error on closing with windows

### PR DESCRIPTION
**Pull request by jlp765**
Win thows an ERROR_INVALID_HANDLE on close.

Doco (possibly) suggests that closing the mapping handle also closes the file handle.

My tests show ignoring the ERROR_INVALID_HANDLE allows repetitive open/close of a memfile

Only tested on win64  

Can someone test on win32 ?

```
import os, memfiles

proc runTst(fname: string) =
  var
    file = memfiles.open(fname, fmRead)
    i = 0
  for field in lines(file):
    inc(i)
  memfiles.close(file)
  echo "Read ",i, " lines!"

proc main() =
  if paramCount() < 1:
    quit("ERROR: Please provide a file name for read testing")
  runTst(paramStr(1))
  runTst(paramStr(1))
main()
```

### CODE REVIEW

**Comment by jlp765** [comment]
Requires further testing

**Comment by jlp765** [comment]
This works for non -d:release

On win64 
-d:release sets the optimization flag (-O3) for gcc and the second open() fails to mapViewOfFileEx()

File: lib/pure/memfiles.nim [hunk 1]
```diff
@@ -247,6 +247,8 @@ proc open*(filename: string, mode: FileMode = fmRead,
       if close(result.handle) == 0:
         result.handle = -1
 
+const ERROR_INVALID_HANDLE = 6
```
Comment (dom96) [inline lib/pure/memfiles.nim]: This should be defined in ``winlean`` (if it isn't already?).

**Comment by dom96** [comment]
Btw, it would also be nice if your PR didn't have 68 commits... but whoever merges this can squash it(!)

**Comment by Araq** [comment]
What's the point in this?

**Comment by jlp765** [comment]
See https://forum.nim-lang.org/t/3270  
The example fails on both windows platforms.

Also, for me both tests\stdlib\tmemfiles1.nim and tests\stdlib\tmemfiles2.nim fail

**Comment by Araq** [comment]
> This works for non -d:release
>
> On win64
> -d:release sets the optimization flag (-O3) for gcc and the second open() fails to mapViewOfFileEx()

So ... is this still true?

**Comment by jlp765** [comment]
Yes, still true.

But I believe it is a different issue (although related).

I haven't been able to solve this second problem.

**Comment by Araq** [comment]
This was fixed differently by me.

**Pull request closed without merging**